doc/source/cluster/key-concepts.rst

```diff
@@ -0,0 +1,50 @@
+.. include:: we_are_hiring.rst
+
+.. _cluster-key-concepts:
+
+Key Concepts
+============
+
+Cluster
+-------
+
+A Ray cluster is a set of one or more nodes that are running Ray and share the same :ref:`head node<cluster-node-types>`.
+
+.. _cluster-node-types:
+
+Node types
+----------
+
+A Ray cluster consists of a :ref:`head node<cluster-head-node>` and a set of :ref:`worker nodes<cluster-worker-node>`.
+
+.. image:: ray-cluster.jpg
+    :align: center
+    :width: 600px
+
+.. _cluster-head-node:
+
+Head node
+~~~~~~~~~
+
+The head node is the first node started by the :ref:`Ray cluster launcher<cluster-launcher>` when trying to launch a Ray cluster. Among other things, the head node hold the :ref:`Global Control Store (GCS)<memory>` and runs the :ref:`autoscaler<cluster-autoscaler>`. Once the head node is started, it will be responsible for launching any additional :ref:`worker nodes<cluster-worker-node>`. The head node itself will also execute tasks and actors to utilize its capacity.
+
+.. _cluster-worker-node:
+
+Worker node
+~~~~~~~~~~~
+
+A worker node is any node in the Ray cluster that is not functioning as head node. Therefore, worker nodes are simply responsible for executing tasks and actors. When a worker node is launched, it will be given the address of the head node to form a cluster.
+
+.. _cluster-launcher:
+
+Cluster launcher
+----------------
+
+The cluster launcher is a process responsible for bootstrapping the Ray cluster by launching the :ref:`head node<cluster-head-node>`. For more information on how to use the cluster launcher, refer to :ref:`cluster launcher CLI commands documentation<cluster-commands>` and the corresponding :ref:`documentation for the configuration file<cluster-config>`.
+
+.. _cluster-autoscaler:
+
+Autoscaler
+----------
```

Comment:
Not really related to this PR, but just bringing up for the visibility -- we should have page of docs explaining how the autoscaler makes scaling decisions using Ray task/actor resource annotations and Ray node resource capacities.

I remember we had posted Ameer's video explaining these things, but
- I can't find the video
- Some written docs would help too

Recently opened an issue https://github.com/ray-project/ray/issues/24323 about this, as users have raised that the docs aren't clear here.

Reply:
Sounds good let's track that separately